cmd/remote-storage/main.go

```diff
@@ -78,7 +80,10 @@ func main() {
 			tm := tenancy.NewManager(&opts.Tenancy)
 			telset := baseTelset // copy
 			telset.Metrics = metricsFactory
-			server, err := app.NewServer(opts, storageFactory, tm, telset)
+
+			traceFactory := v1adapter.NewFactory(storageFactory)
+			depFactory := traceFactory.(depstore.Factory)
```

Comment:
@yurishkuro should we do a runtime cast or do we want to create a new constructor instead?

Reply:
hm, the bigger issue here is that if we want remote-storage to be fully v2 compatible there is not way to configure that, since you have to go through v1 CLI flags. We may need to make a breaking change and switch to yaml config similar to jaeger-v2

Reply:
One thing we perhaps could do is have a subcommand `migrate` which will support all the existing v1 flags (but won't enable them by default on the main command) and will use those flags to populate the config structs and write them out as YAML. But I am not sure this binary is actually used by anyone, so I would start with a breaking change and if people complain we could go this way with `migrate` subcommand.